examples/sat/hillclimber.ipynb

```diff
@@ -0,0 +1,1024 @@
+      "text/plain": [
+       "\u001b[0;31mClass docstring:\u001b[0m\n",
+       "    Find the largest element in a list, while keeping track of query statistics.\n",
+       "    \n",
+       "    Args:\n",
+       "        iterable: iterable to find the maximum in\n",
+       "        default: default value to return if iterable is empty.\n",
+       "        key: function that maps iterable elements to values that are comparable. By default, use the iterable elements.\n",
+       "        error: upper bound on the failure probability of the quantum algorithm.\n",
+       "        stats: object that keeps track of statistics.\n",
+       "    \n",
+       "    Raises:\n",
+       "        ValueError: Raised when the failure rate `error` is not provided and statistics cannot be calculated.\n",
+       "        ValueError: Raised when iterable is an empty sequence and no default is provided.\n",
+       "    \n",
+       "    Returns:\n",
+       "        the desired maximum element\n",
+       "\u001b[0;31mCall docstring:\u001b[0m\n",
+       "    Call self as a function."
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "from qubrabench.algorithms.max import max as qmax\n",
+    "\n",
+    "%pdoc qmax"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": false,
+    "jupyter": {
+     "outputs_hidden": false
+    }
+   },
+   "source": [
+    "## Benchmarking\n",
+    "\n",
+    "The imported `run` function executes the hillclimber on a random instance and returns the statistics in a pandas DataFrame."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {
+    "collapsed": false,
+    "jupyter": {
+     "outputs_hidden": false
+    }
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "\u001b[0;31mClass docstring:\u001b[0m\n",
+       "    External interface to generate weighted sat instances, run the hillclimber algorithm and return statistics.\n",
+       "    \n",
+       "    Args:\n",
+       "        k: Number of literals in a clause\n",
+       "        r: Factor for the number of clauses\n",
+       "        n: size (variable number) of the SAT instances\n",
+       "        n_runs: number of runs to perform in each group\n",
+       "        rng: Source of randomness\n",
+       "        error: Upper bound on the failure rate. Defaults to None.\n",
+       "        random_weights: Optionally providable weights for SAT instance generation. Defaults to None.\n",
+       "        steep: Whether to perform hillclimb steep (greedily). Defaults to False.\n",
+       "    \n",
+       "    Returns:\n",
+       "        Dataframe holding benchmarking statistics of the runs performed.\n",
+       "\u001b[0;31mCall docstring:\u001b[0m\n",
+       "    Call self as a function."
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "%pdoc run"
```

Comment:
why do we just print docstrings without explanation? I'm fine doing this with source code that is sufficiently commented but to me there is not much value in just reading docs (in a notebook).
Do we not care about how they are implemented?

Reply:
I don't mind either way, depends on the target of the notebook.

I assumed it was just to showcase how to run the benchmark and plot the data, but we can also `%psource` and explain anything that's missing.

Reply:
> why do we just print docstrings without explanation? I'm fine doing this with source code that is sufficiently commented but to me there is not much value in just reading docs (in a notebook). Do we not care about how they are implemented?

Yes in fact I think we would like the notebook to be a self-contained tutorial on how to use `qubrabench` to instrument and annotate a certain algorithm, e.g. a hillclimber for SAT solver.

Reply:
Remember - the goal is not to teach people how to use `examples.sat`, but how to use `qubrabench`.

Reply:
Ok yes this makes sense. @ekulos-code, maybe you can update the notebook with this in mind?